data/pc/1.6.4/protocol.json

```diff
@@ -0,0 +1,2286 @@
+            "name": "action",
+            "type": "i16"
+          },
+          {
+            "name": "mode",
+            "type": "i8"
+          },
+          {
+            "name": "item",
+            "type": "slot"
+          }
+        ]
+      ],
+      "packet_set_slot": [
+        "container",
+        [
+          {
+            "name": "windowId",
+            "type": "i8"
+          },
+          {
+            "name": "slot",
+            "type": "i16"
+          },
+          {
+            "name": "item",
+            "type": "slot"
+          }
+        ]
+      ],
+      "packet_window_items": [
+        "container",
+        [
+          {
+            "name": "windowId",
+            "type": "i8"
+          },
+          {
+            "name": "items",
+            "type": [
+              "array",
+              {
+                "countType": "i16",
+                "type": "slot"
+              }
+            ]
+          }
+        ]
+      ],
+      "packet_craft_progress_bar": [
+        "container",
+        [
+          {
+            "name": "windowId",
+            "type": "i8"
+          },
+          {
+            "name": "property",
+            "type": "i16"
+          },
+          {
+            "name": "value",
+            "type": "i16"
+          }
+        ]
+      ],
+      "packet_transaction": [
+        "container",
+        [
+          {
+            "name": "windowId",
+            "type": "i8"
+          },
+          {
+            "name": "action",
+            "type": "i16"
+          },
+          {
+            "name": "accepted",
+            "type": "bool"
+          }
+        ]
+      ],
+      "packet_set_creative_slot": [
+        "container",
+        [
+          {
+            "name": "slot",
+            "type": "i16"
+          },
+          {
+            "name": "item",
+            "type": "slot"
+          }
+        ]
+      ],
+      "packet_enchant_item": [
+        "container",
+        [
+          {
+            "name": "windowId",
+            "type": "i8"
+          },
+          {
+            "name": "enchantment",
+            "type": "i8"
+          }
+        ]
+      ],
+      "packet_update_sign": [
+        "container",
+        [
+          {
+            "name": "position",
+            "type": "position_isi"
+          },
+          {
+            "name": "line1",
+            "type": "string"
+          },
+          {
+            "name": "line2",
+            "type": "string"
+          },
+          {
+            "name": "line3",
+            "type": "string"
+          },
+          {
+            "name": "line4",
+            "type": "string"
+          }
+        ]
+      ],
+      "packet_item_data": [
+        "container",
+        [
+          {
+            "name": "itemType",
+            "type": "i16"
+          },
+          {
+            "name": "itemId",
+            "type": "i16"
+          },
+          {
+            "name": "text",
```

Comment:
This is just raw data in the form of a byte array of the length its prefixed with.